CustomDate - copia.js

```diff
@@ -0,0 +1,127 @@
+class CustomDate {
+    constructor() {
+        this.now = new Date();
+        this.date = new Date();
+        
+        //Use constants to avoid 'magic numbers'
+        this.daysOfTheWeek = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
+        this.HoursInDay = 24;
+        this.MinsInHour = 60;
+        this.SecondsInMin = 60;
+        this.msInSecond = 1000;
+    }
+
+
+    // update this.date object
+    updateDate(newDate) {
+        this.date = newDate;
+    }
+
+
+    // return yesterday's date
+    yesterday() {
+        const yesterday = new Date();
+        yesterday.setDate(now.getDate() - 1);
+        return yesterday;
+    }
+
+
+    // return yesterday's day
+    yesterdayDay() {
+        return this.daysOfTheWeek[(this.now.getDay() - 1) % 7];
+    }
+
+
+    // return tomorrow's date
+    tomorrow() {
+        const tomorrow = new Date();
+        tomorrow.setDate(now.getDate() + 1);
+        return tomorrow;
+    }
+
+
+    // return tomorrow's day
+    tomorrowDay() {
+        return this.daysOfTheWeek[(this.now.getDay() + 1) % 7];
+    }
+
+
+    // add minutes to this.date
+    addMinutes(minutes) {
+        this.date.setMinutes(this.date.getMinutes() + minutes);
+    }
+
+
+    // add hours to this.date
+    addHours(hours) {
+        this.date.setHours(this.date.getHours() + hours);
+    }
+
+
+    // add days to this.date
+    addDays(days) {
+        this.date.setDate(this.date.getDate() + days);
+    }
+
+
+    // Logs the current date-time
+    current() {
+        console.log(this.now);
+    }
+
+
+    // return boolean
+    isFuture() {
+        return this.now < this.date;
```

Comment:
I don't think this code is correct, please check it again

Reply:
I changed it to ValueOf() from docu. and used debugger while runnig it and it give correct output. When comparing the current date with an updated date of future e.g 8-8-2024.